File: src/16.c
#include <stdio.h>
#include <math.h>

int main() {
    // Calculate 2^1000
    double result = pow(2, 1000);

    char str[1000];
    sprintf(str, "%f", result);

    int i = 0;
    int sum = 0;
    while(1){
        if(str[i] == '.'){
            break;
        }
        sum += str[i]-'0';
        i++;
    }
    printf("%i\n",sum);
    return 0;
}

File: src/10-19/16.ml
open Printf

let () =

  let result = 2.0 ** 1000.0 in
  let str = sprintf "%f" result in

  let rec sum_digits i sum =
    if i >= String.length str || str.[i] = '.' then
      sum
    else
      let digit = int_of_char str.[i] - int_of_char '0' in
      sum_digits (i + 1) (sum + digit)
  in

  let sum = sum_digits 0 0 in

  printf "%d\n" sum
